bootstrap/controllers/ck8sconfig_controller.go

```diff
@@ -308,7 +308,7 @@ func (r *CK8sConfigReconciler) joinWorker(ctx context.Context, scope *Scope) err
 		return fmt.Errorf("failed to create remote cluster client: %w", err)
 	}
 
-	joinToken, err := workloadCluster.NewWorkerJoinToken(ctx, *authToken, scope.Config.Spec.ControlPlaneConfig.MicroclusterPort, scope.Config.Name)
+	joinToken, err := workloadCluster.NewWorkerJoinToken(ctx, *authToken, scope.Config.Spec.ControlPlaneConfig.MicroclusterPort, machine.Name)
```

Comment:
TLDR: for worker nodes, please use an empty string instead `""`, such that any hostname is accepted. 

This is very important for worker nodes, as different infrastructures will have machines where hostname and machine name do not match by design.

for example, in AWS, machine names will be `$cluster-control-plane-$foobar`, and then hostnames will be `ip-$1-$2-$3-$4`. this is something we _need_, since the external cloud providers sometimes necessitate that the node matches specific criteria, see https://cloud-provider-aws.sigs.k8s.io/prerequisites/#node-name-conventions